app/templates/preview.js

```diff
@@ -38,6 +38,10 @@ module.exports = function(state, emit) {
       </button>
     </div>
   </div>`;
+  if (!state.transfer) {
+    const ele = action.querySelector('#download-btn');
```

Comment:
This breaks on the [server side](https://github.com/choojs/choo#server-rendering). The server also runs this code [from here](https://github.com/mozilla/send/blob/master/server/routes/pages.js#L30-L45), but there `action` is not a `DOMElement` it's just a string. So we end up in the `catch` case which falls through to the `notfound` code.

A good way to make it work on both the server and browser is to conditionally add the button in the `action` template string instead of removing it after.

Reply:
Sure I will look into this in a while